types/priv_validator.go

```diff
@@ -95,6 +95,12 @@ func (pv MockPV) SignVote(ctx context.Context, chainID string, vote *tmproto.Vot
 		return err
 	}
 	vote.Signature = sig
+	if len(vote.Extension) > 0 {
```

Comment:
Same two comments here:
* We decided to sign 0-length vote extensions as well
* We need to sign the "canonical extension", rather than just the extension's bytes

Reply:
We now sign over the canonicalized vote extension, whether there's an extension present or not.